servers/src/main/java/tachyon/master/permission/FileSystemPermissionChecker.java

```diff
@@ -49,7 +50,63 @@ public static synchronized void initializeFileSystem(boolean permissionCheckEnab
   }
 
   /**
-   * This method checks access permission.
+   * Checks requested permission on the last node and basic permission on the whole path.
+   *
+   * @param user who requests access permission
+   * @param groups in which user belongs to
+   * @param action requested {@link FileSystemAction} by user
+   * @param inodes all the inodes retrieved by traversing the path
+   * @throws IOException if permission checking fails
+   */
+  public static void checkSelfPermission(String user, List<String> groups, FileSystemAction action,
```

Comment:
`self` means the last inode in the path. Compared to the parent inode of the path.

Remove this word in updated patch to avoid confusion.